tools/oxide-bar/src/main.rs

```diff
@@ -276,7 +313,7 @@ fn get_x11rb_events(
     }
 }
 
-fn get_state(event_sender_mutex: Arc<Mutex<Sender<EventType>>>) {
+fn thread_state(event_sender_mutex: Arc<Mutex<Sender<EventType>>>) {
```

Comment:
could you explain, why there is another thread started inside this function, which already runs in a seperate thread ?:)
couldn't you write a function state_signal_channel_async that blocks, until a state is received and then returns ?:)

Reply:
This is an intermediate thread that awaits a state signal from the oxide-ipc and then transforms this into a oxide-bar datastructure. 
This is done in order to not need to implement oxide-bar specific datastructures into oxide ipc.
I think the performance hit is very minimal because each of the threads waits on the other one and therefore will need little to no runtime